lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -1238,6 +1265,76 @@ def running(self):
                 return True
         return False
 
+    def _compare_dict_allow_more_present(self, av, bv):
+        '''
+        Compare two dictionaries for whether every entry of the first is in the second.
+        '''
+        for key, value in av.items():
+            if key not in bv:
+                return False
+            if bv[key] != value:
```

Comment:
Hmm, there is no definition what `KEY: null` should mean for `env`. There are only interpretions :) For me, `KEY: null` is not a valid value. Also, `docker` might die (return an error) if you pass `None` as the value of an environment variable (or it might ignore it). Allowing `null` / `None` would be a new feature anyway.